lms/static/js/search/collections/SearchCollection.js

```diff
@@ -0,0 +1,88 @@
+RequireJS.define([
+    'backbone',
+    'js/search/models/SearchResult'
+], function (Backbone, SearchResult) {
+    'use strict';
+
+    return Backbone.Collection.extend({
+
+        model: SearchResult,
+        pageSize: 20,
+        totalCount: 0,
+        accessDeniedCount: 0,
+        searchTerm: '',
+        page: 0,
+        url: '/search/',
+        fetchXhr: null,
+
+        initialize: function (models, options) {
+            // call super constructor
+            Backbone.Collection.prototype.initialize.apply(this, arguments);
+            if (options && options.course_id) {
+                this.url += options.course_id;
+            }
+        },
+
+        performSearch: function (searchTerm) {
+            this.fetchXhr && this.fetchXhr.abort();
+            this.searchTerm = searchTerm || '';
+            this.totalCount = 0;
+            this.accessDeniedCount = 0;
+            this.page = 0;
+            this.fetchXhr = this.fetch({
+                data: {
+                    search_string: searchTerm,
+                    page_size: this.pageSize,
+                    page_index: 0
+                },
+                type: 'POST',
+                success: function (self, xhr) {
+                    self.trigger('search');
```

Comment:
Seems odd that you're completely ignoring the data that you're receiving from the server. Where does the search handler get this data?